contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
-        if (s.sarcophagi[sarcoId].resurrectionTime <= block.timestamp) {
+        // Confirm resurrection time has not yet passed
+        if (sarcophagus.resurrectionTime <= block.timestamp) {
             revert LibErrors.SarcophagusIsUnwrappable();
         }
 
-        // Confirm that the new resurrection time is in the future
+        // Confirm that new resurrection time is in future
         if (resurrectionTime <= block.timestamp) {
             revert LibErrors.NewResurrectionTimeInPast(resurrectionTime);
         }
 
-
-        // Confirm that the new resurrection time doesn't exceed the sarcophagus's maximumRewrapInterval
-        if (resurrectionTime > block.timestamp + s.sarcophagi[sarcoId].maximumRewrapInterval) {
+        // Confirm that new resurrection time doesn't exceed sarcophagus's maximumRewrapInterval
+        if (resurrectionTime > block.timestamp + sarcophagus.maximumRewrapInterval) {
             revert LibErrors.NewResurrectionTimeTooLarge(resurrectionTime);
         }
 
-        // For each archaeologist on the sarcophagus, transfer their digging fee allocations to them
-        address[] memory bondedArchaeologists = s
-            .sarcophagi[sarcoId]
-            .archaeologists;
-
+        // track total digging fees across all archaeologists on the sarcophagus
         uint256 totalDiggingFees = 0;
 
-        for (uint256 i = 0; i < bondedArchaeologists.length; i++) {
-            // Get the archaeolgist's fee data
-            LibTypes.ArchaeologistStorage memory archaeologistData = LibUtils
-                .getArchaeologist(sarcoId, bondedArchaeologists[i]);
-
-            // Transfer the archaeologist's digging fee allocation to the archaeologist's reward pool
-            s.archaeologistRewards[bondedArchaeologists[i]] += archaeologistData.diggingFee;
+        // pay digging fee to each cursed archaeologist on the sarcophagus
+        address[] storage archaeologistAddresses = sarcophagus.cursedArchaeologistAddresses;
+        for (uint256 i = 0; i < archaeologistAddresses.length; i++) {
+            LibTypes.CursedArchaeologist storage cursedArchaeologist = sarcophagus
+                .cursedArchaeologists[archaeologistAddresses[i]];
 
-            // Add to the total of digging fees paid
-            archaeologistData.diggingFeesPaid += archaeologistData.diggingFee;
-
-            // Add the archaeologist's digging fee to the sum
-            totalDiggingFees += archaeologistData.diggingFee;
-
-            // Update the archaeologist's data in storage
-            s.sarcophagusArchaeologists[sarcoId][
-                bondedArchaeologists[i]
-            ] = archaeologistData;
+            // transfer digging fee to archaeologist's reward pool
+            // todo: consider adding this amount to archaeologistProfile.freeBond instead
+            s.archaeologistRewards[archaeologistAddresses[i]] += cursedArchaeologist.diggingFee;
```

Comment:
Consider combining `archaeologistProfile.freeBond` and `s.archaeologistRewards` so that archaeologists can minimize their costs when withdrawing funds and we aren't keeping unnecessary data structures

Reply:
We might have considered this at some point before, iirc, and I think the argument was that the archaeologist will want to be able keep track of their earnings on the system. This way they won't have to remember how much of their free bond pool came from their pockets in order to know how much they have earned.